modules/gapi/src/backends/onnx/gonnxbackend.cpp

```diff
@@ -143,6 +143,39 @@ class ONNXCompiled {
     void run();
 };
 
+void appendExecutionProvider(Ort::SessionOptions          *session_options,
+                             const cv::gapi::onnx::ep::EP &execution_provider) {
+    namespace ep = cv::gapi::onnx::ep;
+    switch (execution_provider.index()) {
+        case ep::EP::index_of<ep::OpenVINO>(): {
+             const auto &ovep = cv::util::get<ep::OpenVINO>(execution_provider);
+             OrtOpenVINOProviderOptions options;
+
+             options.device_id = ovep.device_id.c_str();
+
+             options.enable_vpu_fast_compile = ovep.enable_vpu_fast_compile;
+             options.context = ovep.context;
+             options.enable_opencl_throttling = ovep.enable_opencl_throttling;
+             options.enable_dynamic_shapes = ovep.enable_dynamic_shapes;
+
+             if (ovep.device_type) {
+                options.device_type = ovep.device_type->c_str();
+             }
+             if (ovep.cache_dir) {
+                options.cache_dir = ovep.cache_dir->c_str();
+             }
+             if (ovep.num_of_threads) {
+                options.num_of_threads = *ovep.num_of_threads;
+             }
+
+             session_options->AppendExecutionProvider_OpenVINO(options);
```

Comment:
This thing might throw, make sense to wrap it and show the understandable error message...